packages/commercetools-api-request-builder/src/create-service.js

```diff
@@ -92,18 +92,27 @@ export default function createService (
     // Pass some options to further configure the URI:
     // - `projectKey`: will prefix the URI with the given projectKey
     build (options: BuildOptions = {}): string {
+      // FIXME: make it an option of `createRequestBuilder`
       const { projectKey } = options
 
       const queryParams = buildQueryString(this.params)
 
       const uri =
         (projectKey ? `/${projectKey}` : '') +
         endpoint +
-        (this.params.id ? `/${this.params.id}` : '') +
+        getIdOrKey(this.params) +
         (queryParams ? `?${queryParams}` : '')
 
       setDefaultParams.call(this)
       return uri
     },
   })
 }
+
+function getIdOrKey (params: Object): string {
```

Comment:
This looks odd, why not two functions and put the `if - else` logic where needed?